### ISSUE TITLE
feat(workload): add normalized exponential arrival sampler for inference-perf fidelity

### ISSUE THREAD

**Issue by susiejojo** [inference-sim/inference-sim#845]
## Problem Statement

BLIS cannot replicate inference-perf's "constant" load generation behavior, causing systematic bias in training data. Ground-truth experiments collected with kubernetes-sigs/inference-perf use **normalized exponential inter-arrival times** that span the exact configured duration, while BLIS uses **truncated Poisson arrivals** that end earlier than the target duration.

### Real Example
From `training/trainval_data/20260217-155451-llama-2-7b-tp1-codegen/profile.yaml`:

```json
{
  "load": {
    "stages": [
      {"duration": 600, "rate": 5},
      {"duration": 600, "rate": 10}
    ],
    "type": "constant"
  }
}
```

**Expected behavior:**
- Stage 1: 3000 requests (5 × 600) spanning 0-600s
- Stage 2: 6000 requests (10 × 600) spanning 600-1200s
- **Total: 9000 requests spanning exactly 1200.0 seconds**

### How inference-perf Implements This

From [`inference_perf/loadgen/load_timer.py`](https://github.com/kubernetes-sigs/inference-perf/blob/main/inference_perf/loadgen/load_timer.py#L48-L61):

```python
class ConstantLoadTimer(LoadTimer):
    def start_timer(self, initial: Optional[float] = None) -> Generator[float, None, None]:
        num_requests = int(self._rate * self._duration)
        
        # Generate random intervals using EXPONENTIAL distribution
        intervals = self._rand.exponential(1 / self._rate, num_requests)
        
        # Normalize intervals to sum EXACTLY to duration
        total_interval_time = np.sum(intervals)
        scale_factor = self._duration / total_interval_time
        normalized_intervals = intervals * scale_factor
        
        # Yield request times
        next_time = time.monotonic() if initial is None else initial
        for interval in normalized_intervals:
            next_time += interval
            yield next_time
```

**Key properties:**
1. Pre-generates **all** intervals before dispatch
2. Uses exponential distribution (same as Poisson) 
3. **Normalizes** intervals to sum exactly to duration
4. Result: N requests with exponential spacing spanning exactly T seconds
5. Coefficient of variation: **CV < 1.0** (constrained variance)

### Code Flow

1. **`training/run_blis_and_compute_loss.py`** computes expected count:
   ```python
   total_requests = sum(round(s["rate"] * s["duration"]) for s in stages_config)
   spec = {
       "num_requests": total_requests,  # 9000
       "inference_perf": {...}
   }
   ```

2. **`sim/workload/inference_perf.go:124,168`** hardcodes Poisson:
   ```go
   clients = append(clients, ClientSpec{
       Arrival: ArrivalSpec{Process: "poisson"},  // ← Hardcoded
       // ...
   })
   ```

3. **`sim/workload/generator.go:247-256`** generates incrementally:
   ```go
   for currentTime < horizon {
       iat := arrivalSampler.SampleIAT(rng)  // Exponential, unconstrained
       currentTime += iat
       if currentTime >= horizon {
           break  // ← Stops at horizon boundary
       }
       // Generate request...
   }
   ```

4. **`sim/workload/generator.go:322-325`** truncates to exact count:
   ```go
   // Truncate to maxRequests after merge-sort
   if maxRequests > 0 && int64(len(allRequests)) > maxRequests {
       allRequests = allRequests[:maxRequests]
   }
   ```

### What Happens

For the example above (rate=5, duration=600s, target=3000 requests):

1. Generator samples exponential IATs until `currentTime >= 600s`
2. Due to Poisson variance, generates ~3054 requests by 600s (√3000 ≈ 55 stdev)
3. Truncates to exactly 3000 requests
4. **Last request arrives at ~588s** (12 seconds lost)
5. Coefficient of variation: **CV = 1.0** (unconstrained Poisson)

### Missing: Pre-Generation with Normalization

BLIS's architecture generates requests **incrementally** (streaming), controlled by:
1. **Horizon** (time boundary): `currentTime < horizon`
2. **maxRequests** (count cap): post-generation truncation

There is **no mechanism** to:
- Pre-generate N intervals
- Normalize them to sum to exactly T seconds
- Preserve the exponential distribution shape while constraining the sum

### Existing Arrival Samplers

From `sim/workload/arrival.go`:

| Sampler | Inter-Arrival Time | CV | Request Count | Duration |
|---------|-------------------|-----|---------------|----------|
| `PoissonSampler` | `Exponential(1/λ)` | 1.0 | Approximate (N ± √N) | Exact (T) |
| `ConstantArrivalSampler` | `1/λ` | 0.0 | Approximate | Exact (T) |
| `GammaSampler` | `Gamma(α, β)` | Configurable | Approximate | Exact (T) |
| `WeibullSampler` | `Weibull(k, λ)` | Configurable | Approximate | Exact (T) |

**None support:** Exponential IAT + exact count + exact duration simultaneously.

### Systematic Bias

Ground-truth data:
- ✅ Requests span full duration (0-600s)
- ✅ Late-stage load observable (590-600s has requests)
- ✅ Stage transitions at exact boundaries (600.0s)

BLIS simulation:
- ❌ Requests span shortened duration (0-588s)
- ❌ Late-stage load missing (590-600s has no requests)
- ❌ Stage transitions blurred (truncation artifacts)

### Quantitative Difference

For a typical stage (rate=10, duration=600s, target=6000 requests):

**inference-perf normalized exponential:**
- Mean IAT: 100.0ms
- Std IAT: ~99.9ms (normalized CV ≈ 0.999)
- Duration: exactly 600.000s
- Count: exactly 6000

**BLIS truncated Poisson:**
- Mean IAT: 100.0ms  
- Std IAT: 100.0ms (CV = 1.0)
- Duration: ~588s (2% shorter)
- Count: exactly 6000 (post-truncation)

**Lost coverage:** ~12s per 600s stage = **2% of measurement window**

### Training Implications

The training pipeline (`run_blis_and_compute_loss.py`) compares:
- Ground truth: requests distributed 0-600s
- BLIS predictions: requests distributed 0-588s

The loss metric (RMSE of APE) includes error from:
1. ✅ Latency model accuracy (what we want to measure)
2. ❌ **Workload generation mismatch** (confounding variable)

Late-arriving requests in ground truth have higher latency (queue buildup). BLIS never predicts these because it doesn't generate them.

## Proposed Solution

Add a `NormalizedExponentialSampler` to `sim/workload/arrival.go`:

### High-Level Design

```go
type NormalizedExponentialSampler struct {
    intervals []int64  // Pre-generated normalized intervals
    index     int      // Current position
}

func NewNormalizedExponentialSampler(rate float64, duration int64, count int) *NormalizedExponentialSampler {
    // 1. Generate count exponential samples
    raw := make([]float64, count)
    for i := range raw {
        raw[i] = rng.ExpFloat64() / rate
    }
    
    // 2. Normalize to sum exactly to duration
    sum := 0.0
    for _, v := range raw {
        sum += v
    }
    scale := float64(duration) / sum
    
    // 3. Convert to microseconds
    intervals := make([]int64, count)
    for i, v := range raw {
        intervals[i] = int64(v * scale)
    }
    
    return &NormalizedExponentialSampler{intervals: intervals}
}

func (s *NormalizedExponentialSampler) SampleIAT(_ *rand.Rand) int64 {
    if s.index >= len(s.intervals) {
        return 1  // Defensive: should never happen
    }
    iat := s.intervals[s.index]
    s.index++
    return iat
}
```

### Integration Points

1. **`sim/workload/inference_perf.go`**: Pass duration and request count to arrival spec
   ```go
   Arrival: ArrivalSpec{
       Process: "normalized_exponential",
       Duration: &stageDurationUs,
       Count: &stageRequestCount,
   }
   ```

2. **`sim/workload/generator.go`**: Check for pre-generated mode
   ```go
   // Skip horizon loop for pre-generated samplers
   if sampler.IsPreGenerated() {
       for i := 0; i < sampler.Count(); i++ {
           iat := sampler.SampleIAT(clientRNG)
           currentTime += iat
           // Generate request...
       }
   }
   ```

### Alternative: Extend Existing Samplers

Could add a `Normalized` flag to `ArrivalSpec` that wraps any sampler:
```go
type ArrivalSpec struct {
    Process    string
    Normalized bool    // Pre-generate and normalize to exact duration
    Count      *int64  // Required when Normalized=true
}
```

## Related Work

- Issue #481: Inference-perf multi-turn mapping (fixed request count, didn't address timing)
- H30-H32 experiments: Discovered 2% throughput variance attributed to "Poisson noise"
- `docs/plans/h30-h32-replay-plan.md:63`: "±2.5% measures Poisson noise, not step-time accuracy"

## Acceptance Criteria

1. BLIS can generate workloads matching inference-perf's normalized exponential timing
2. For ground-truth profile with rate=R, duration=D:
   - Request count: exactly R×D
   - Duration span: exactly D seconds (last arrival ≈ D)
   - Inter-arrival distribution: exponential with normalized sum
3. Training loss metric no longer includes workload mismatch confound
4. Backward compatible: existing Poisson specs unchanged

## References

- inference-perf source: https://github.com/kubernetes-sigs/inference-perf/blob/main/inference_perf/loadgen/load_timer.py
- Ground-truth data: `training/trainval_data/*/profile.yaml`
- BLIS arrival samplers: `sim/workload/arrival.go`
